scrapy/pqueues.py

```diff
@@ -63,21 +52,36 @@ def __init__(self, crawler, downstream_queue_cls, key, startprios=()):
         self.curprio = None
         self.init_prios(startprios)
 
+    @classmethod
+    def bring_prios_up_to_date(_, startprios):
+
+        def _is_valid(x):
+            return isinstance(x, (tuple, list)) and len(x) == 2
+
+        is_valid = all(map(_is_valid, startprios))
+        if is_valid:
+            return startprios
+
+        return [(x, None) for x in startprios]
+
     def init_prios(self, startprios):
```

Comment:
I’m not 100% sure, since quite some time has passed since I reviewed this pull request the last time and I’m not 100% sure of what has changed since then, but I think the fact that `startprios` is a different structure now may be a backward compatible change that, if unavoidable, we should at least document. I don’t think `bring_prios_up_to_date` makes up for it, since subclasses implementing their own `init_prios` are unlikely to be using `super`.

Reply:
Format changed only for `ScrapyPriorityQueue`, not every pq in the wild.